upgrade/util/util.go

```diff
@@ -243,6 +277,55 @@ func CheckUpgradePathSupported(namespace string, lhClient lhclientset.Interface)
 	return nil
 }
 
+// checkEngineUpgradePathSupported returns error if the upgrade path from
+// current EngineImage version to emeta.ControllerAPIVersion is not supported.
+func checkEngineUpgradePathSupported(namespace string, lhClient lhclientset.Interface, newEngineVersion emeta.VersionOutput) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, "failed checking Engine upgrade path")
+	}()
+
+	engineImageVersions, err := GetCurrentReferredLonghornEngineImageVersions(namespace, lhClient)
+	if err != nil {
+		return err
+	}
+
+	if len(engineImageVersions) == 0 {
+		return nil
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"newEngineControllerAPIVersion":    newEngineVersion.ControllerAPIVersion,
+		"newEngineControllerAPIMinVersion": newEngineVersion.ControllerAPIMinVersion,
+		"newEngineClientAPIVersion":        newEngineVersion.CLIAPIVersion,
+		"newEngineClientAPIMinVersion":     newEngineVersion.CLIAPIMinVersion,
+	}).Infof("Checking if the engine upgrade path from %+v is supported", engineImageVersions)
+
+	checkSupportVersion := func(currentVersion, newVersion, newMinVersion int) error {
+		if currentVersion > newVersion {
+			return fmt.Errorf("downgrading from %v to %v is not supported", currentVersion, newVersion)
+		}
```

Comment:
note: this just means we did not do for forward compatibility for longhorn manager if we have a newer engine version to commute with.